packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -407,33 +462,41 @@ export class SharedConditionalsTest extends AbstractConditionalsTest {
 
     this.assertText('T1F2');
   }
+}
 
-  ['@test it maintains DOM stability when condition changes from a truthy to a different truthy value']() {
-    this.renderValues(true);
-
-    this.assertText('T1');
+export class SharedSyntaxConditionalsTest extends SharedConditionalsTest {
+  ['@htmlbars it does not update when the unbound helper is used']() {
+    let template = `${
+      this.templateFor({ cond: '(unbound cond1)', truthy: 'T1', falsy: 'F1' })
+    }${
+      this.templateFor({ cond: '(unbound cond2)', truthy: 'T2', falsy: 'F2' })
+    }`;
 
-    this.takeSnapshot();
+    this.render(template, { cond1: true, cond2: false });
 
-    this.runTask(() => set(this.context, 'cond1', 'hello'));
+    this.assertText('T1F2');
 
-    this.assertText('T1');
+    this.runTask(() => this.rerender());
 
-    this.assertInvariants();
-  }
+    this.assertText('T1F2');
 
-  ['@test it maintains DOM stability when condition changes from a falsy to a different falsy value']() {
-    this.renderValues(false);
+    this.runTask(() => set(this.context, 'cond1', false));
 
-    this.assertText('F1');
+    this.assertText('T1F2');
 
-    this.takeSnapshot();
+    this.runTask(() => {
+      set(this.context, 'cond1', true);
+      set(this.context, 'cond2', true);
+    });
 
-    this.runTask(() => set(this.context, 'cond1', ''));
+    this.assertText('T1F2');
 
-    this.assertText('F1');
+    this.runTask(() => {
+      set(this.context, 'cond1', true);
+      set(this.context, 'cond2', false);
+    });
 
-    this.assertInvariants();
+    this.assertText('T1F2');
   }
 
   ['@test it tests for `isTruthy` on the context if available']() {
```

Comment:
We should probably have a version for this for the helper version too. Can you add that?

While you are at that, I noticed this test doesn't test both starting conditions. Can you try to copy the format of the test immediately above us to test both starting conditions in parallel?